src/libsyntax/codemap.rs

```diff
@@ -592,16 +592,21 @@ impl CodeMap {
         CodeMap {
             files: RefCell::new(Vec::new()),
             expansions: RefCell::new(Vec::new()),
-            file_loader: file_loader
+            file_loader: RefCell::new(file_loader)
         }
     }
 
+    pub fn set_file_loader(&self, file_loader: Box<FileLoader>) {
+        let mut loader = self.file_loader.borrow_mut();
+        *loader = file_loader;
+    }
+
```

Comment:
Glad to see someone made use of this before I got back to my IDE-like project!
Any specific reason why you need to swap the file loader after initialization?